zebra-network/src/protocol/external/codec.rs

```diff
@@ -590,20 +599,38 @@ impl Codec {
         Ok(Message::Mempool)
     }
 
-    fn read_filterload<R: Read>(&self, mut _reader: R) -> Result<Message, Error> {
-        return Err(Error::Parse("filterload messages are not implemented"));
-    }
+    fn read_filterload(&self, mut reader: Cursor<&BytesMut>) -> Result<Message, Error> {
+        const MAX_FILTER_LENGTH: usize = 36000;
+        const FILTERLOAD_REMAINDER_LENGTH: usize = 4 + 4 + 1;
+
+        let filter_length: usize = (reader.get_ref().len() - FILTERLOAD_REMAINDER_LENGTH)
+            .try_into()
+            .unwrap();
```

Comment:
Where does the `try_into`/`unwrap` come from? If the LHS and RHS are both `usize`, their difference should be `usize` also, right?

Reply:
I wonder if we should first check that `body_len >= FILTERLOAD_REMAINDER_LENGTH` and that `body_len <= FILTERLOAD_REMAINDER_LENGTH + MAX_FILTER_LENGTH` before we do any subtractions, so we know that `filter_length = body_len - FILTERLOAD_REMAINDER_LENGTH` cannot have underflowed, and then we have its exact value so we don't need to do a `min` call later. WDYT?

Reply:
re: `try_into` I think this is a relic from when I was using `Read::take` which wanted a u64